lib/src/components/action-icon/ActionIcon.tsx

```
@@ -195,12 +196,24 @@ const StyledButton = styled('button', {
   ]
 })
 
+const ConditionallyWrapWithTooltip = ({ hideTooltip, label, children }) => {
```

Instead of false conditionals can we have `showTooltip` with a default being false?

---

The default would be to show the tooltip not hide it. I had it like that until I had to use it to hide some of the tooltips and I believe that 
`<ActionIcon hideTooltip><Icon/></ActionIcon>` reads better than `<ActionIcon showTooltip={false}><Icon/></ActionIcon>`

---

I don't think I've ever seen a regular HTML element which has a negative attribute before which is why I don't like having negative attributes in our components. 
However, yes `showTooltip={false}`  isn't preferable, which is why I suggested it defaults to false.
But yes, then it doesn't default to `true` - so people will need to use it when they need it, so there will be a learning curve.
But is that really bad?

It really all comes back to the same thing of Tooltips being obfuscated and combined into ActionIcon in a way I'm not convinced about.
On the other hand - I do see how it will remove multiple lines of code from atom-core and developers wont even need to think about it so 🤷 .

---

So you have concerns on the way I have combined the 2? it would help if you could share what way you think it is better

---

Hey, I'll have another look at this tomorrow and have a think

---

The component we expose to the consumer (`ActionIcon`, not `ConditionallyWrapWithTooltip`) does default the bool to `false`, right?

I share @LimeWub's preference for false defaults on boolean props, and  while I think I have a slight preference for `showTooltip` over `hideTooltip`, I think this strikes a good balance. If the design decision is that tooltips should be the rule rather than the exception, facilitating that feels more important than worrying about whether `hide` or `show` feels more natural. (I also think that design decision is a more important factor than saving 4 lines of code anyway.)

I guess the "correct" way to do this in terms of component composition is probably to have the `TooltipActionIcon` from `atom-core` in here as a separate component, and try to make using that, rather than using `ActionIcon` our default behaviour. But for me personally I can't see a benefit to that which outweighs the faff.

---

if we definitely want to use `showTooltip` could you not set a default value for it and instead set it as an optional boolean, then in the component have something like `const shouldShowTooltip = showTooltip !== false` which would pick up `true` and `undefined`.

I do agree with Gary though I think there's a balance to be had, and that 'hideX` isn't abhorrent by default

---

ok so we have 2 open issues to address:

- Should the Tooltip be part of the ActionIcon or have 2 components? 
Where I sit with this is that we have tried for 18 months now, and having two components isn't working, as people will use the wrong one. I feel that the separation isn't worth the pestering effort. 

I think that if we want all of the ActionIcons to show the tooltip (which is something will reconfirm with Miguel) then why offer the plain one and create the opportunity for misuse? We could have 2 of them and only expose for usage the one with the tooltip externally and use the one without internally for places like the Inputs etc, but again that will only depend if we don't want to show the tooltip there

---

@kerryambt, sorry I kind of feel that `const shouldShowTooltip = showTooltip !== false` is more complex than just setting the default to true here

---

I find it rare to think of an icon or icon button that should not have a tooltip to explain its purpose, I think we should make it the default and consistent use

---

Thanks for the feedback @MiguelGDLR,

@LimeWub @hollg @kerryambt Based on this, what would be the preferred solution now? Should I have two components and only expose the Tooltip wrapped one for external usage? or just have one

---

I'm going to be picking this up next week, but I'm thinking of going with `showTooltip` and then default it to true. I think this will address some of the naming concerns and keep the behaviour close to the real life need.

---

Regarding the name, I won't push it, but it doesn't address the naming concern.

`showTooltip` starts with a verb, which is usually used for functions. `hasTooltip` it starts with a word that says "this is a boolean"